[1.6] - Setuptools version

When I install an instance from scratch, it does not work anymore without setting the version of setuptools.

A few weeks ago, I did not need to do that, because when `make` creates the virtual environment, it retrieves the latest setuptool version (which, right now, would be 19.4, and used to be 18.2).

Without specifiying that `setuptools==18.2` in my project's `setup.py` file, the "pip" process crashes.

I have a (hudge...) stack trace of that (and I do not really get what is going wrong...). Am I the only one to have this problem ?
```shell
Running setup.py install for PasteScript
    Skipping installation of D:\Applications\Mapfish\prepublication16\.build\venv\Lib\site-packages\paste\__init__.py (namespace package)
    Searching for PasteDeploy
    Reading https://pypi.python.org/simple/PasteDeploy/
    Best match: PasteDeploy 1.5.2
    Downloading https://pypi.python.org/packages/source/P/PasteDeploy/PasteDeploy-1.5.2.tar.gz#md5=352b7205c78c8de4987578d19431af3b
    Processing PasteDeploy-1.5.2.tar.gz
    Writing d:\applications\cygwin64\tmp\easy_install-pa9xyp\PasteDeploy-1.5.2\setup.cfg
    Running PasteDeploy-1.5.2\setup.py -q bdist_egg --dist-dir d:\applications\cygwin64\tmp\easy_install-pa9xyp\PasteDeploy-1.5.2\egg-dist-tmp-wv7bmo
    Installed d:\applications\cygwin64\tmp\easy_install-pa9xyp\pastedeploy-1.5.2\.eggs\paste-2.0.2-py2.7.egg
    installing library code to build\bdist.win32\egg
    removing 'build\bdist.win32\egg' (and everything under it)
    Extracting pastedeploy-1.5.2-py2.7.egg to d:\applications\cygwin64\tmp\pip-build-404la2\pastescript\.eggs
    Installed d:\applications\cygwin64\tmp\pip-build-404la2\pastescript\.eggs\pastedeploy-1.5.2-py2.7.egg
    Searching for Paste>=1.3
    Reading https://pypi.python.org/simple/Paste/
    Best match: Paste 2.0.2
    Downloading https://pypi.python.org/packages/source/P/Paste/Paste-2.0.2.tar.gz#md5=4bfc8a7eaf858f6309d2ac0f40fc951c
    Processing Paste-2.0.2.tar.gz
    Writing d:\applications\cygwin64\tmp\easy_install-pf9_t1\Paste-2.0.2\setup.cfg
    Running Paste-2.0.2\setup.py -q bdist_egg --dist-dir d:\applications\cygwin64\tmp\easy_install-pf9_t1\Paste-2.0.2\egg-dist-tmp-fx8894
    Installed d:\applications\cygwin64\tmp\easy_install-pf9_t1\paste-2.0.2\.eggs\pastedeploy-1.5.2-py2.7.egg
    installing library code to build\bdist.win32\egg
    removing 'build\bdist.win32\egg' (and everything under it)
    Extracting paste-2.0.2-py2.7.egg to d:\applications\cygwin64\tmp\pip-build-404la2\pastescript\.eggs
    Installed d:\applications\cygwin64\tmp\pip-build-404la2\pastescript\.eggs\paste-2.0.2-py2.7.egg
    Installing D:\Applications\Mapfish\prepublication16\.build\venv\Lib\site-packages\PasteScript-1.7.5-py2.7-nspkg.pth
    Installing paster-script.py script to D:\Applications\Mapfish\prepublication16\.build\venv\Scripts
    Installing paster.exe script to D:\Applications\Mapfish\prepublication16\.build\venv\Scripts
    Installing paster.exe.manifest script to D:\Applications\Mapfish\prepublication16\.build\venv\Scripts
  Running setup.py install for PasteDeploy
    Searching for Paste>=1.3
    Reading https://pypi.python.org/simple/Paste/
    Best match: Paste 2.0.2
    Downloading https://pypi.python.org/packages/source/P/Paste/Paste-2.0.2.tar.gz#md5=4bfc8a7eaf858f6309d2ac0f40fc951c
    Processing Paste-2.0.2.tar.gz
    Writing d:\applications\cygwin64\tmp\easy_install-t8qk7m\Paste-2.0.2\setup.cfg
    Running Paste-2.0.2\setup.py -q bdist_egg --dist-dir d:\applications\cygwin64\tmp\easy_install-t8qk7m\Paste-2.0.2\egg-dist-tmp-treyfm
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "d:\applications\cygwin64\tmp\pip-build-404la2\PasteDeploy\setup.py", line 58, in <module>
        """
      File "D:\Applications\Python27\Lib\distutils\core.py", line 151, in setup
        dist.run_commands()
      File "D:\Applications\Python27\Lib\distutils\dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "D:\Applications\Python27\Lib\distutils\dist.py", line 972, in run_command
        cmd_obj.run()
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\install.py", line 61, in run
        return orig.install.run(self)
      File "D:\Applications\Python27\Lib\distutils\command\install.py", line 563, in run
        self.run_command('build')
      File "D:\Applications\Python27\Lib\distutils\cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "D:\Applications\Python27\Lib\distutils\dist.py", line 972, in run_command
        cmd_obj.run()
      File "D:\Applications\Python27\Lib\distutils\command\build.py", line 127, in run
        self.run_command(cmd_name)
      File "D:\Applications\Python27\Lib\distutils\cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "D:\Applications\Python27\Lib\distutils\dist.py", line 972, in run_command
        cmd_obj.run()
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\build_py.py", line 52, in run
        self.build_package_data()
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\build_py.py", line 107, in build_package_data
        for package, src_dir, build_dir, filenames in self.data_files:
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\build_py.py", line 65, in __getattr__
        self.data_files = self._get_data_files()
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\build_py.py", line 78, in _get_data_files
        self.analyze_manifest()
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\build_py.py", line 127, in analyze_manifest
        self.run_command('egg_info')
      File "D:\Applications\Python27\Lib\distutils\cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "D:\Applications\Python27\Lib\distutils\dist.py", line 972, in run_command
        cmd_obj.run()
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\egg_info.py", line 177, in run
        ep.require(installer=installer)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 2385, in require
        items = working_set.resolve(reqs, env, installer)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 827, in resolve
        dist = best[req.key] = env.best_match(req, ws, installer)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 1072, in best_match
        return self.obtain(req, installer)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 1084, in obtain
        return installer(requirement)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\dist.py", line 380, in fetch_build_egg
        return cmd.easy_install(req)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\easy_install.py", line 640, in easy_install
        return self.install_item(spec, dist.location, tmpdir, deps)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\easy_install.py", line 670, in install_item
        dists = self.install_eggs(spec, download, tmpdir)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\easy_install.py", line 853, in install_eggs
        return self.build_and_install(setup_script, setup_base)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\easy_install.py", line 1081, in build_and_install
        self.run_setup(setup_script, setup_base, args)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\easy_install.py", line 1067, in run_setup
        run_setup(setup_script, args)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 229, in run_setup
        with setup_context(setup_dir):
      File "D:\Applications\Python27\Lib\contextlib.py", line 17, in __enter__
        return self.gen.next()
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 195, in setup_context
        yield
      File "D:\Applications\Python27\Lib\contextlib.py", line 35, in __exit__
        self.gen.throw(type, value, traceback)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 166, in save_modules
        saved_exc.resume()
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 141, in resume
        six.reraise(type, exc, self._tb)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 154, in save_modules
        yield saved
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 194, in setup_context
        __import__('setuptools')
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\__init__.py", line 11, in <module>
        from setuptools.extern.six.moves import filterfalse, map
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\extern\__init__.py", line 1, in <module>
        from pkg_resources.extern import VendorImporter
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 3130, in <module>
        @_call_aside
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 3116, in _call_aside
        f(*args, **kwargs)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 3155, in _initialize_master_working_set
        add_activation_listener(lambda dist: dist.activate())
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 956, in subscribe
        callback(dist)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 3155, in <lambda>
        add_activation_listener(lambda dist: dist.activate())
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 2655, in activate
        declare_namespace(pkg)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 2225, in declare_namespace
        _handle_ns(packageName, path_item)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 2194, in _handle_ns
        path.sort(key=sort_key)
      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 2192, in sort_key
        return sys_path.index(_normalize_cached(os.sep.join(parts)))
    ValueError: 'd:\\applications\\cygwin64\\tmp\\easy_install-t8qk7m\\paste-2.0.2' is not in list
    Complete output from command D:\Applications\Mapfish\prepublication16\.build\venv\Scripts\python.exe -c "import setuptools, tokenize;__file__='d:\\applications\\cygwin64\\tmp\\pip-build-404la2\\PasteDeploy\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record d:\applications\cygwin64\tmp\pip-iinrhe-record\install-record.txt --single-version-externally-managed --compile --install-headers D:\Applications\Mapfish\prepublication16\.build\venv\include\site\python2.7:
    running install

    running build

    running build_py

    creating build

    creating build\lib

    creating build\lib\paste

    copying paste\__init__.py -> build\lib\paste

    creating build\lib\paste\deploy

    copying paste\deploy\compat.py -> build\lib\paste\deploy

    copying paste\deploy\config.py -> build\lib\paste\deploy

    copying paste\deploy\converters.py -> build\lib\paste\deploy

    copying paste\deploy\loadwsgi.py -> build\lib\paste\deploy

    copying paste\deploy\paster_templates.py -> build\lib\paste\deploy

    copying paste\deploy\util.py -> build\lib\paste\deploy

    copying paste\deploy\__init__.py -> build\lib\paste\deploy

    running egg_info

    Searching for Paste>=1.3

    Reading https://pypi.python.org/simple/Paste/

    Best match: Paste 2.0.2

    Downloading https://pypi.python.org/packages/source/P/Paste/Paste-2.0.2.tar.gz#md5=4bfc8a7eaf858f6309d2ac0f40fc951c

    Processing Paste-2.0.2.tar.gz

    Writing d:\applications\cygwin64\tmp\easy_install-t8qk7m\Paste-2.0.2\setup.cfg

    Running Paste-2.0.2\setup.py -q bdist_egg --dist-dir d:\applications\cygwin64\tmp\easy_install-t8qk7m\Paste-2.0.2\egg-dist-tmp-treyfm

    Traceback (most recent call last):

      File "<string>", line 1, in <module>

      File "d:\applications\cygwin64\tmp\pip-build-404la2\PasteDeploy\setup.py", line 58, in <module>

        """

      File "D:\Applications\Python27\Lib\distutils\core.py", line 151, in setup

        dist.run_commands()

      File "D:\Applications\Python27\Lib\distutils\dist.py", line 953, in run_commands

        self.run_command(cmd)

      File "D:\Applications\Python27\Lib\distutils\dist.py", line 972, in run_command

        cmd_obj.run()

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\install.py", line 61, in run

        return orig.install.run(self)

      File "D:\Applications\Python27\Lib\distutils\command\install.py", line 563, in run

        self.run_command('build')

      File "D:\Applications\Python27\Lib\distutils\cmd.py", line 326, in run_command

        self.distribution.run_command(command)

      File "D:\Applications\Python27\Lib\distutils\dist.py", line 972, in run_command

        cmd_obj.run()

      File "D:\Applications\Python27\Lib\distutils\command\build.py", line 127, in run

        self.run_command(cmd_name)

      File "D:\Applications\Python27\Lib\distutils\cmd.py", line 326, in run_command

        self.distribution.run_command(command)

      File "D:\Applications\Python27\Lib\distutils\dist.py", line 972, in run_command

        cmd_obj.run()

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\build_py.py", line 52, in run

        self.build_package_data()

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\build_py.py", line 107, in build_package_data

        for package, src_dir, build_dir, filenames in self.data_files:

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\build_py.py", line 65, in __getattr__

        self.data_files = self._get_data_files()

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\build_py.py", line 78, in _get_data_files

        self.analyze_manifest()

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\build_py.py", line 127, in analyze_manifest

        self.run_command('egg_info')

      File "D:\Applications\Python27\Lib\distutils\cmd.py", line 326, in run_command

        self.distribution.run_command(command)

      File "D:\Applications\Python27\Lib\distutils\dist.py", line 972, in run_command

        cmd_obj.run()

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\egg_info.py", line 177, in run

        ep.require(installer=installer)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 2385, in require

        items = working_set.resolve(reqs, env, installer)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 827, in resolve

        dist = best[req.key] = env.best_match(req, ws, installer)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 1072, in best_match

        return self.obtain(req, installer)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 1084, in obtain

        return installer(requirement)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\dist.py", line 380, in fetch_build_egg

        return cmd.easy_install(req)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\easy_install.py", line 640, in easy_install

        return self.install_item(spec, dist.location, tmpdir, deps)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\easy_install.py", line 670, in install_item

        dists = self.install_eggs(spec, download, tmpdir)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\easy_install.py", line 853, in install_eggs

        return self.build_and_install(setup_script, setup_base)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\easy_install.py", line 1081, in build_and_install

        self.run_setup(setup_script, setup_base, args)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\command\easy_install.py", line 1067, in run_setup

        run_setup(setup_script, args)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 229, in run_setup

        with setup_context(setup_dir):

      File "D:\Applications\Python27\Lib\contextlib.py", line 17, in __enter__

        return self.gen.next()

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 195, in setup_context

        yield

      File "D:\Applications\Python27\Lib\contextlib.py", line 35, in __exit__

        self.gen.throw(type, value, traceback)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 166, in save_modules

        saved_exc.resume()

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 141, in resume

        six.reraise(type, exc, self._tb)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 154, in save_modules

        yield saved

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\sandbox.py", line 194, in setup_context

        __import__('setuptools')

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\__init__.py", line 11, in <module>

        from setuptools.extern.six.moves import filterfalse, map

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\setuptools\extern\__init__.py", line 1, in <module>

        from pkg_resources.extern import VendorImporter

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 3130, in <module>

        @_call_aside

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 3116, in _call_aside

        f(*args, **kwargs)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 3155, in _initialize_master_working_set

        add_activation_listener(lambda dist: dist.activate())

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 956, in subscribe

        callback(dist)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 3155, in <lambda>

        add_activation_listener(lambda dist: dist.activate())

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 2655, in activate

        declare_namespace(pkg)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 2225, in declare_namespace

        _handle_ns(packageName, path_item)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 2194, in _handle_ns

        path.sort(key=sort_key)

      File "D:\Applications\Mapfish\prepublication16\.build\venv\lib\site-packages\pkg_resources\__init__.py", line 2192, in sort_key

        return sys_path.index(_normalize_cached(os.sep.join(parts)))

    ValueError: 'd:\\applications\\cygwin64\\tmp\\easy_install-t8qk7m\\paste-2.0.2' is not in list

    ----------------------------------------
    Command "D:\Applications\Mapfish\prepublication16\.build\venv\Scripts\python.exe -c "import setuptools, tokenize;__file__='d:\\applications\\cygwin64\\tmp\\pip-build-404la2\\PasteDeploy\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record d:\applications\cygwin64\tmp\pip-iinrhe-record\install-record.txt --single-version-externally-managed --compile --install-headers D:\Applications\Mapfish\prepublication16\.build\venv\include\site\python2.7" failed with error code 1 in d:\applications\cygwin64\tmp\pip-build-404la2\PasteDeploy
CONST_Makefile:699: recipe for target '.build/requirements.timestamp' failed
make: *** [.build/requirements.timestamp] Error 1
```

--- a/c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl
+++ b/c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl
@@ -19,6 +19,8 @@ PRINT_VERSION ?= 3
 
 PIP_CMD ?= $(VENV_BIN)/pip
 PIP_INSTALL_ARGS += install --trusted-host pypi.camptocamp.net
+PIP_VERSION ?= pip>=6
+SETUPTOOL_VERSION ?= setuptools>=12
 
 GIT_MODULES_FOLDER ?= .git/modules/
 
@@ -691,7 +693,7 @@ $(VENV_BIN)/flake8: .build/dev-requirements.timestamp
 	virtualenv --setuptools --no-site-packages .build/venv
 	$(PIP_CMD) install \
 		--index-url http://pypi.camptocamp.net/pypi \
-		'pip>=6,<7' 'setuptools>=12'
+		'$(PIP_VERSION)' '$(SETUPTOOL_VERSION)'
 	touch $@
 
 .build/requirements.timestamp: $(EGGS_DEPENDENCIES)

